interfaces/datastructure/Mask2D.h

```diff
@@ -0,0 +1,165 @@
+     * instanceId, the instance Id of the detected object, if instanceId < 0, it means that the current pixel belongs to a "stuff" class which is uncountable, otherwise it belongs to a "thing" class
+     * confidence, confidence score between 0 and 1, the confidence score of the segmentation, if confidence < 0, it means that the confidence score is not available
+     */
+    struct SegInfo {
+        SegInfo() = default;
+        SegInfo(int16_t c, int16_t i, float cf) : classId(c), instanceId(i), confidence(cf) {}
+        int16_t classId = -1;
+        int16_t instanceId = -1;
+        float confidence = -1.f;
+
+        friend class boost::serialization::access;
+        template<typename Archive>
+        void serialize(Archive &ar, const unsigned int version) {
+            ar& classId;
+            ar& instanceId;
+            ar& confidence;
+        }
+    };
+    using MaskInfoType = std::map<uint8_t, SegInfo>;
+    using ClassLabelType = std::map<int16_t, std::string>;
+
+    /// @brief default constructor
+    Mask2D() = default;
+
+    /// @brief constructor with args
+    /// @param[in] id ID of the Mask2D
+    /// @param[in] type segmentation type
+    /// @param[in] mask mask
+    /// @param[in] info mask info
+    /// @param[in] label mapping from class ID to label
+    Mask2D(uint32_t id, Segmentation2DType type, SRef<Image> mask, const MaskInfoType& info, const ClassLabelType& label);
+
+    /// @brief default destructor
+    ~Mask2D() = default;
+
+    /// @brief set Id
+    /// @param[in] id Mask2D ID
+    void setId(uint32_t id);
+
+    /// @brief set mask
+    /// @param[in] mask pointer to mask buffer
+    void setMask(SRef<Image> mask);
+
+    /// @brief set mask info
+    /// @param[in] maskInfo mask info used to interpret the mask
+    void setMaskInfo(const MaskInfoType& maskInfo);
+
+    /// @brief set mapping from class ID to label (string)
+    /// @param[in] classIdToLabel mapping from class ID to label
+    void setClassLabels(const ClassLabelType& classIdToLabel);
+
+    /// @brief set segmentation type
+    /// @param[in] type segmentation type
+    void setSegmentationType(Segmentation2DType type);
+
+    /// @brief get Id
+    /// @return ID of the Mask2D object
+    uint32_t getId() const;
+
+    /// @brief get mask
+    /// @return pointer to mask
+    SRef<Image> getMask() const;
+
+    /// @brief get const mask
+    /// @return pointer to const mask
+    SRef<const Image> getConstMask() const;
+
+    /// @brief get mask info
+    /// @return mask info
+    const MaskInfoType& getMaskInfo() const;
+
+    /// @brief get labels of classes
+    /// @return labels of classes
+    const ClassLabelType& getClassLabels() const;
+
+    /// @brief get segmentation type
+    /// @return segmentation type
+    Segmentation2DType getSegmentationType() const;
+
+    /// @brief print info
+    void print() const;
```

Comment:
Maybe print could be replaced by a toString() method, so that other component can decide of to print it.

But you can keep it if you want since thes is probably going to change soon with the work on telemetry.